### PR TITLE
Solve discrepancy between docs and code, making end async.

### PR DESCRIPTION
**Pull request by Kristof-Mattei**
There is a discrepancy between the docs about whether `Pool.end()` is `async` or not.

The docs say: YES. 
https://github.com/postgres-pool/postgres-pool/blame/d2b1995d91d4a20fdfea5ff0772fb9095c30badd/README.md#L73

The code says: NO. 
https://github.com/postgres-pool/postgres-pool/blob/7c2397449c8bc60369ae56885712fc537f493004/src/index.ts#L436

I took a stab at making `end()` async.

It doesn't CHANGE anything in the code except for that we now explicitly await to `end` a `poolClient`.

Since we use this library quite extensively in our private code I took the liberty of compiling it and using it in our test-suite, and apart from having to add `await` in front of `pool.end()` all tests pass identically. 

The advantage now is that I can call `await pool.end()` and be sure that once that's done all connections are closed, something which we couldn't before, as we didn't await the `client.end()`.

[Or remove the `await` from the docs 😉 ](https://github.com/postgres-pool/postgres-pool/pull/56)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -541,19 +542,24 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
...
     };
 
-    client.errorHandler = (err: Error): void => {
-      this._removeConnection(client);
+    // we assign an async promise to a void
+    // because we don't want to enforce a promise signature
+    // on errorHandler
+    // also, EventEmitter.on("error") doesn't support () => Promise<void> (see the on("error"))
+    client.errorHandler = async (err: Error): Promise<void> => {
+      await this._removeConnection(client);
```
Comment (Kristof-Mattei) [inline src/index.ts:562]: I do wonder if it is needed that we remove the connection here. If `postgres-pool` uses the `client`, then it'll `release()` in the `finally` block:

https://github.com/postgres-pool/postgres-pool/pull/57/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R467-R469

If the user acquires the connection, they are mandated to call `.release()` themselves!

Comment (jgeurts) [reply src/index.ts:562]: I believe I originally followed how pg-pool handled errors - they terminate and remove the client from the pool. I think that might be ok to leave as the behavior as a catch-all.

File: src/index.ts [hunk 2]
```diff
@@ -541,19 +542,24 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
...
       }
     };
 
-    client.errorHandler = (err: Error): void => {
-      this._removeConnection(client);
+    // we assign an async promise to a void
+    // because we don't want to enforce a promise signature
+    // on errorHandler
+    // also, EventEmitter.on("error") doesn't support () => Promise<void> (see the on("error"))
+    client.errorHandler = async (err: Error): Promise<void> => {
```
Comment (jgeurts) [inline src/index.ts:561]: Thoughts on leaving errorHandler return void?

```ts
client.errorHandler = (err: Error): void => {
  this._removeConnection(client).finally(() => this.emit('error', err, client));
}
```

File: src/index.ts [hunk 3]
```diff
@@ -677,7 +683,7 @@ export class Pool extends (EventEmitter as new () => PoolEmitter) {
       this.connections.splice(connectionIndex, 1);
     }
 
-    client.end().catch((ex) => {
+    await client.end().catch((ex) => {
```
Comment (jgeurts) [inline src/index.ts:686]: Can you please change this to a try/catch block?
```ts
try {
  await client.end();
} catch (ex) {
  const { message } = ex as Error;
  if (!/This socket has been ended by the other party/giu.test(message)) {
    this.emit('error', ex);
  }
}
```

Comment (Kristof-Mattei) [reply src/index.ts:686]: Will do.

File: src/index.ts (hunk 1 above)
Comment (Kristof-Mattei) [reply src/index.ts:562]: Will leave it in!

File: src/index.ts (hunk 2 above)
Comment (Kristof-Mattei) [reply src/index.ts:561]: Makes sense if we want to keep `this._removeConnection`. Will do.

**Pull request merged**